zipline/_protocol.pyx

```diff
@@ -508,6 +508,9 @@ cdef class BarData:
             # asset isn't alive
             return False
 
+        if asset.auto_close_date and dt >= asset.auto_close_date:
```

Comment:
Maybe just me, but I'm not sure when a consumer should use is_alive_for_session or check the auto-close or both.